packages/core/src/autoscaling/autoscaled_pool.ts

```diff
@@ -640,6 +670,19 @@ export class AutoscaledPool {
 
         betterClearInterval(this.autoscaleInterval);
         betterClearInterval(this.maybeRunInterval);
+        if (this.tasksDonePerSecondInterval) betterClearInterval(this.tasksDonePerSecondInterval);
         if (this.snapshotter) await this.snapshotter.stop();
     }
+
+    protected async _incrementTasksDonePerSecond(intervalCallback: () => void) {
```

Comment:
seems there's no need for this to be `async`, even the `intervalCallback` is synchronous 🙂